ls1intum/Artemis
# Fix exam quiz submissions not being saved properly

### Checklist

### Motivation and Context
This resolves #1770 

### Description
- force save submission when the user clicks on save
- implement can deactivate
- add TODO for generic message
- small code refactoring


### Steps for Testing
Prerequisite: An Exam was properly set up with multiple choice questions and is running.
- Check that in the exam view, a warning is displayed when you try to reload the page.
- If the network connection is lost and you try to save ...
    * by waiting 60 seconds after a change
    * by manually clicking on "Safe and Continue" _(this will force a save, regardless of)_
    * by switching the exercise in the navigation bar
    
    a warning shows up, that there submission could not be saved.
- if there is no sign of it being not saved, it is present after reload *or* the end of the exam (summary)

### Screenshots
